kotlinx-coroutines-core/common/src/Yield.kt

```diff
@@ -19,7 +19,9 @@ public suspend fun yield(): Unit = suspendCoroutineUninterceptedOrReturn sc@ { u
     val context = uCont.context
     context.checkCompletion()
     val cont = uCont.intercepted() as? DispatchedContinuation<Unit> ?: return@sc Unit
-    if (!cont.dispatcher.isDispatchNeeded(context)) {
+    // Special case for Unconfined dispatcher that does not support the concept of "dispatch" and where
+    // "dispatchYield" function cannot be used.
+    if (!cont.dispatcher.isDispatchSupported(context)) {
```

Comment:
If it is internal and applicable only for `Unconfined` dispatcher and `isDispatchSupported` is internal anyway, maybe we should compare `cont.dispatcher` to it instead of adding one more public method?

Reply:
Got rid of the method, added equality comparison to dispatcher. Also, fixed a problem where yielding from a wrapped dispatcher around Unconfined would crash.

Reply:
Alternatively, both methods can being replaced by a single one with enumeration: `NEEDED`, `SUPPORTED`, `UNAVAILABLE`.

Reply:
@fvasco  That would be a binary API change that would break 3rd party dispatcher implementations. It is easier to special-case `Unconfined` here since we don't really have used-cases at the table for user-defined dispatchers with unconfined-like behavior.